### PR TITLE
Create "matchedFields" section in QueryResult

### PR DESCRIPTION
**Pull request by eddturner**
This uses a terms query.

A lot of code needed refactoring for this, due to duplication in the way queries were handled. All queries now use a `SolrRequest` object to specify what is wanted in a request. `SolrRequestConverter` then will convert it to either a Spring query or a SolrQuery. Only SolrQuery handles queries with terms.

I think the easiest way to review this is to look at `SolrRequestConverter` and `SolrRequest` and see where it is used.

### CODE REVIEW

File: common-rest/src/main/java/uk/ac/ebi/uniprot/api/rest/respository/RepositoryConfig.java [hunk 1]
```diff
@@ -62,6 +63,12 @@ public SolrClient uniProtSolrClient(HttpClient httpClient, RepositoryConfigPrope
         }
     }
 
+    @Bean
+    @Profile("live")
```
[the remark this answers is not in the document]
Comment (LeonardoGonzales) [reply common-rest/src/main/java/uk/ac/ebi/uniprot/api/rest/respository/RepositoryConfig.java]: I saw bellow.. :-)

File: common-rest/src/main/java/uk/ac/ebi/uniprot/api/common/repository/search/SolrRequest.java [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Represents a request object containing the details to create a query to send to Solr.
+ *
+ * @author Edd
+ */
+@Data
+@ToString
```
[the remark this answers is not in the document]
Comment (eddturner) [reply common-rest/src/main/java/uk/ac/ebi/uniprot/api/common/repository/search/SolrRequest.java]: awesome thanks!

File: common-rest/src/main/java/uk/ac/ebi/uniprot/api/common/repository/search/SolrQueryRepository.java [hunk 3]
```diff
@@ -28,70 +29,75 @@
...
 
-    public QueryResult<T> searchPage(SimpleQuery query, String cursor, Integer pageSize) {
+    public QueryResult<T> searchPage(SolrRequest request, String cursor, Integer pageSize) {
         if (pageSize == null || pageSize <= 0) {
             pageSize = DEFAULT_PAGE_SIZE;
         }
         try {
             CursorPage page = CursorPage.of(cursor, pageSize);
-            QueryResponse solrResponse = solrTemplate.execute(getSolrCursorCallback(query, page.getCursor(), pageSize));
+            requestConverter.toSolrQuery(request);
```
[the remark this answers is not in the document]
Comment (eddturner) [reply common-rest/src/main/java/uk/ac/ebi/uniprot/api/common/repository/search/SolrQueryRepository.java]: gosh! yes you're right, i think this was an accident ... thank you and well spotted!

File: common-rest/src/main/java/uk/ac/ebi/uniprot/api/common/repository/search/term/TermInfoConverter.java [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+                    .stream()
+                    .map(termEntryToTermInfo())
+                    .filter(Utils::nonNull)
+                    .collect(Collectors.toList());
+        } else {
+            return emptyList();
+        }
+    }
+
+    private Function<Map.Entry<String, List<TermsResponse.Term>>, TermInfo> termEntryToTermInfo() {
```
[the remark this answers is not in the document]
Comment (eddturner) [reply common-rest/src/main/java/uk/ac/ebi/uniprot/api/common/repository/search/term/TermInfoConverter.java]: great -- done :)

**Pull request merged**